earth_enterprise/src/common/khMTTypes.h

```diff
@@ -148,9 +151,173 @@ class khMTSet
     khLockGuard guard(mutex);
     return std::vector<T>(set.begin(), set.end());
   }
+  void clear(void) {
+    khLockGuard guard(mutex);
+    set.clear();
+  }
+  size_t size(void) {
+    khLockGuard guard(mutex);
+    return set.size();
+  }
 };
 
 
+template <class K, class V>
+class khMTMap
+{
+ protected:
+  typedef std::unordered_map<K,V> Map;
+  Map     map;
+  mutable std::mutex mutex;
+
+ private:
+  // private and unimplimented
+  khMTMap& operator=(const khMTMap&);
+
+ public:
+  khMTMap(void) { }
+
+  void insert(const std::pair<K,V> &val) {
+    std::lock_guard<std::mutex> guard(mutex);
+    (void)map.insert(val);
+  }
+  void erase(const K &val) {
+    std::lock_guard<std::mutex> guard(mutex);
+    if (map.find(val) != map.end()) {
```

Comment:
Is this check necessary? I assume just calling `map.erase` will include this check?